opm/simulators/flow/BlackoilModelNldd.hpp

```diff
@@ -656,6 +656,7 @@ class BlackoilModelNldd {
                 } else if (res[ii] > tol[ii]) {
                     report.setReservoirFailed({types[ii], CR::Severity::Normal, compIdx});
                 }
+                report.setReservoirConvergenceMetric(types[ii], compIdx, res[ii]);
```

Comment:
This will unconditionally create a `numComp * 2` array of [`ReservoirConvergenceMetric`](https://github.com/OPM/opm-simulators/blob/f04f4c5765e4fcdca619230b8bc455995722a15c/opm/simulators/timestepping/ConvergenceReport.hpp#L66) objects.  Do we ever collect these `ConvergenceReport` objects for multiple domains and/or report steps?  If so, that additional memory could add up.  If not, i.e., if we're only ever looking at the report for a single domain at a time, then this is fine.

Reply:
This is only for a single domain at a time. The object gets recreated every iteration, and returned up the stack until solveDomainGaussSeidel() where it is used. It dies at the end of that function.